automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/methodconfigs/WorkspaceMethodConfigDetailsPage.scala

```diff
@@ -1,274 +1,180 @@
-        val fieldInputQuery: Query = xpath(s"//*[@data-test-id='$field-text-input']/..//input")
-        searchField(fieldInputQuery).value = expression
-      }
-    }
-
-    def checkSaveButtonState:String = {
-      val button = await enabled saveEditedMethodConfigButtonQuery
-      button.attribute("data-test-state").getOrElse("")
-    }
-
-    def clickSaveButton() = {
-      val button = await enabled saveEditedMethodConfigButtonQuery
-      // The button can sometimes scroll off the page and become unclickable. Therefore we need to scroll it into view.
-      webDriver.asInstanceOf[JavascriptExecutor].executeScript("arguments[0].scrollIntoView(true)", button.underlying)
-      click on button
-    }
+  def checkSaveButtonState(): String = {
+    saveEditedMethodConfigButton.getState
+  }
 
-    def cancelEdits() = {
-      click on (await enabled cancelEditMethodConfigModeButtonQuery)
-    }
+  def saveEdits(expectSuccess: Boolean = true): Unit = {
+    // The button can sometimes scroll off the page and become unclickable. Therefore we need to scroll it into view.
+    saveEditedMethodConfigButton.scrollToVisible()
+    saveEditedMethodConfigButton.doClick()
+    if (expectSuccess)
+      openLaunchAnalysisModalButton.awaitVisible()
+  }
 
-    def isLaunchAnalysisButtonPresent() = {
-      await enabled openLaunchAnalysisModalButtonQuery
-      find(openLaunchAnalysisModalButtonQuery).size == 1
-    }
+  def editMethodConfig(newName: Option[String] = None, newSnapshotId: Option[Int] = None, newRootEntityType: Option[String] = None,
+                       inputs: Option[Map[String, String]] = None, outputs: Option[Map[String, String]] = None): Unit = {
+    openEditMode()
+    await spinner "Loading attributes..."
 
-    def clickLaunchAnalysisButtonError(): ErrorModal = {
-      click on (await enabled openLaunchAnalysisModalButtonQuery)
-      new ErrorModal
-    }
+    if (newName.isDefined) { editMethodConfigNameInput.setText(newName.get) }
+    if (newSnapshotId.isDefined) { changeSnapshotId(newSnapshotId.get) }
+    if (newRootEntityType.isDefined) { editMethodConfigRootEntityTypeSelect.select(newRootEntityType.get)}
+    if (inputs.isDefined) { changeInputsOutputs(inputs.get) }
+    if (outputs.isDefined) { changeInputsOutputs(outputs.get)}
 
-    def verifyMethodConfigurationName(methodConfigName: String) = {
-      await enabled methodConfigNameTextQuery
+    saveEdits()
+  }
 
-      val methodConfigNameElement = find(methodConfigNameTextQuery)
-      methodConfigNameElement.get.text == methodConfigName
+  def changeSnapshotId(newSnapshotId: Int): Unit = {
+    editMethodConfigSnapshotIdSelect.select(newSnapshotId.toString)
+  }
 
+  private def changeInputsOutputs(fields: Map[String, String]): Unit = {
+    for ((field, expression) <- fields) {
+      val fieldInputQuery: Query = xpath(s"//*[@data-test-id='$field-text-input']/..//input")
+      searchField(fieldInputQuery).value = expression
     }
```

Comment:
i got confused as to how this works for output fields. then i realized that input here is refering to the text input field. not sure if this would benefit from some comments about that